tools/decode-packets/src/main/scala/DecodePackets.scala

```diff
@@ -0,0 +1,194 @@
+import java.io.{File, FileWriter, BufferedWriter}
+import java.nio.charset.CodingErrorAction
+import net.psforever.packet.PacketCoding
+import scodec.bits._
+import scodec.Attempt.{Failure, Successful}
+import java.nio.file.{Files, Paths, StandardCopyOption}
+import scala.io.{Codec, Source}
+import util.control.Breaks._
+import scala.collection.parallel.CollectionConverters._
+import scopt.OParser
+import sys.process._
+
+case class Config(
+    outDir: String = System.getProperty("user.dir"),
+    preprocessed: Boolean = false,
+    skipExisting: Boolean = false,
+    files: Seq[File] = Seq()
+)
+
+object DecodePackets {
+  def main(args: Array[String]): Unit = {
+
+    val builder = OParser.builder[Config]
+
+    val parser = {
+      import builder._
+      OParser.sequence(
+        programName("psf-decode-packets"),
+        opt[String]('o', "out-dir")
+          .action((x, c) => c.copy(outDir = x))
+          .text("Output directory"),
+        opt[Unit]('p', "preprocessed")
+          .action((x, c) => c.copy(preprocessed = true))
+          .text("Files are already preprocessed gcapy ascii files (do not call gcapy)"),
+        opt[Unit]('s', "skip-existing")
+          .action((x, c) => c.copy(skipExisting = true))
+          .text("Skip files that already exist in out-dir"),
+        arg[File]("<file>...")
+          .unbounded()
+          .required()
+          .action((x, c) => c.copy(files = c.files :+ x))
+      )
+    }
+
+    val opts = OParser.parse(parser, args, Config()) match {
+      case Some(config) =>
+        config
+      case _ =>
+        sys.exit(1)
+    }
+
+    val outDir = new File(opts.outDir);
+    if(!outDir.exists()) {
+      outDir.mkdirs()
+    } else if(outDir.isFile()) {
+      println(s"error: out-dir is file")
+      sys.exit(1)
+    }
+
+     opts.files.foreach { file =>
+      if(!file.exists) {
+        println(s"file ${file.getAbsolutePath()} does not exist")
+        sys.exit(1)
+      }
+    }
+
+    val tmpFolder = new File(System.getProperty("java.io.tmpdir") + "/psf-decode-packets")
```

Comment:
Since this will be buried away from the user in some system temporary folder is it worth clearing contents of that folder if it exists before starting a run, in the event the previous run failed for some reason?

I'm not sure what the default behavour of FileWriter is when the file already exists